### PR TITLE
feat(client) ts-migrate AppMountNotifier Component

### PR DESCRIPTION
**Pull request by sharmaabhinav**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: client/src/components/AppMountNotifier.tsx [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
   bindActionCreators({ appMount }, dispatch);
 
-class AppMountNotifier extends Component {
+type MyProps = {
+  appMount: () => void;
+  render: () => ReactNode;
+};
+
+class AppMountNotifier extends Component<MyProps> {
```
Comment (ShaunSHamilton) [inline client/src/components/AppMountNotifier.tsx:16]:
```suggestion
type AppMountNotifierProps = {
  appMount: () => void;
  render: () => ReactNode;
};

class AppMountNotifier extends Component<AppMountNotifierProps> {
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Hello there. Thank you, for taking the time to contribute.

We would appreciate general better naming conventions. We do not have much of a guide for this _yet_ . The main reason for this is when we are using text search, it makes it easier to find things, when the names are unique. Thank you, again.

File: client/src/components/AppMountNotifier.tsx (hunk 1 above)
Comment (sharmaabhinav) [reply client/src/components/AppMountNotifier.tsx]: Thanks for point out . Fixed

**Comment by ShaunSHamilton** [comment]
Hey there,

Thank you for taking the time to contribute to the TypeScript migration.

The main issue has been updated a lot over the last few days, and so we suggest you re-read it and update this pull request to match if you have the time.

The main goals for a PR are:

- [ ] Changing the filename from PascalCase to kebab-case and ensuring that the imports in code are all still valid.
- [x] Changing the file extension to .tsx (if a file has React JSX syntax)
- [x] Fixing as many TypeScript and Eslint errors as you can while refactoring the code.
If you need any assistance, please reach out in our contributing chat room.

**Comment by sharmaabhinav** [comment]
> Hey there,
> 
> Thank you for taking the time to contribute to the TypeScript migration.
> 
> The main issue has been updated a lot over the last few days, and so we suggest you re-read it and update this pull request to match if you have the time.
> 
> The main goals for a PR are:
> 
> * [x]  Changing the filename from PascalCase to kebab-case and ensuring that the imports in code are all still valid.
> * [x]  Changing the file extension to .tsx (if a file has React JSX syntax)
> * [x]  Fixing as many TypeScript and Eslint errors as you can while refactoring the code.
>   If you need any assistance, please reach out in our contributing chat room.

Took care of file name change also

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Type packages should not be installed in the root directory. These changes can be reverted, as the package is already installed on `next`.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Pull request merged**